server/src/test/java/com/octopus/teamcity/opentelemetry/server/LogMaskerTest.java

```diff
@@ -0,0 +1,42 @@
+package com.octopus.teamcity.opentelemetry.server;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.Test;
+
+import java.util.Locale;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class LogMaskerTest {
```

Comment:
Consider these cases:
- multiline string, and break the API key over the newline
- spaces or other nasty characters either side of the key

Reply:
Also consider [data-driven tests](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-ValueSource) here.

Reply:
added new tests to commit 8229d75